src/main/java/seedu/address/model/order/predicates/OrderNamePredicate.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.order.predicates;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.util.PredicateUtil;
+import seedu.address.model.customer.Customer;
+import seedu.address.model.order.Order;
+import seedu.address.model.util.predicate.SingleFieldPredicate;
+
+/**
+ * Predicate for filtering orders by their customers' names.
+ */
+public class OrderNamePredicate extends SingleFieldPredicate<Order> {
+
+    public static final String MESSAGE_CONSTRAINTS = "Customer name keywords must not be empty.";
+
+    private final List<Customer> customerList;
+
+    /**
+     * Creates a predicate to filter orders by the orders' customers' names.
+     * The customer list is needed here to access the orders' customers' information.
+     */
+    public OrderNamePredicate(List<String> keywords, List<Customer> customerList) {
+        super(keywords);
+        this.customerList = customerList;
+    }
+
+    @Override
+    public double getSimilarityScore(Order order) {
+        String phone = getPhoneByOrder(order);
+        requireNonNull(phone);
+        return PredicateUtil.getWordSimilarityScoreIgnoreCase(phone, getKeywords());
+    }
+
+    @Override
+    public boolean test(Order order) {
+        String phone = getPhoneByOrder(order);
+        requireNonNull(phone);
+        return PredicateUtil.containsPrefixWordIgnoreCase(phone, getKeywords());
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return (other instanceof SingleFieldPredicate)
+                && super.equals(other)
+                && this.customerList.equals(((OrderNamePredicate) other).customerList);
+    }
+
+    private String getPhoneByOrder(Order order) {
```

Comment:
this not needed here? seems like a repeat in `OrderPhonePredicate`

Reply:
mb, forgot to change the methods' and variables' names after copy-pasting - fixed